util/flightcontrol/flightcontrol_test.go

```diff
@@ -223,6 +223,28 @@ func TestContention(t *testing.T) {
 	wg.Wait()
 }
 
+func TestMassiveParallel(t *testing.T) {
+	var retryCount int64
+	g := &Group[string]{}
+	eg, ctx := errgroup.WithContext(context.Background())
+	for i := 0; i < 1000; i++ {
+		eg.Go(func() error {
+			_, err := g.Do(ctx, "key", func(ctx context.Context) (string, error) {
+				return "", errors.Errorf("always fail")
+			})
+			if errors.Is(err, errRetryTimeout) {
+				atomic.AddInt64(&retryCount, 1)
+			}
+			return err
```

Comment:
Returning `nil` in here changes timing for the test so that it almost never produces the timeout anymore.